lib/kubernetes/events/kubernetes_event_monitor.rb

```diff
@@ -0,0 +1,40 @@
+class KubernetesEventMonitor
+  def initialize(api_endpoint, api_version)
+    @api_endpoint = api_endpoint
+    @api_version = api_version
+  end
+
+  def inventory
+    require 'kubeclient'
+    @inventory ||= Kubeclient::Client.new(@api_endpoint, @api_version)
+  end
+
+  def watcher(version = nil)
+    @watcher ||= inventory.watch_events(version)
+  end
+
+  def start
+    trap(:TERM) { $kube_log.info('EventMonitor#start: ignoring SIGTERM') }
+    @inventory = nil
+    @watcher = nil
+  end
+
+  def stop
+    watcher.finish
```

Comment:
Actually it really doesn't matter because even if you set it to nil it would reuse @api_endpoint that is impossible to change after creation of the object.